githubUtils.js

```diff
@@ -0,0 +1,199 @@
+    const msgsCommits = await getCommitMessages(project.repoName, project.oldSha);
+    const strCommits = msgsCommits.join("\n");
+    const projectName = project.repoName.toUpperCase();
+    return `${projectName}\n\n${strCommits}`;
+  });
+  logs = await Promise.all(logs);
+
+  logs = logs.join("\n\n");
+  if (await isNotLatestManifestsVersion()) {
+    logs = `⚠️ **The production version of manifests is behind the latest staging version. Consider upgrading to the latest version before merging this pull request.** \n\n ${logs}`;
+  }
+
+  if (await isNotLatestTerraformVersion()) {
+    logs = `⚠️ **The production version of the Terraform infrastructure is behind the latest staging version. Consider upgrading to the latest version before merging this pull request.** \n\n ${logs}`;
+  }
+  return logs;
+}
+
+const getCommitMessages = async (repo, sha) => {
+  const { data: commits } = await octokit.repos.listCommits({
+    owner: GH_CDS,
+    repo,
+    per_page: 50,
+  });
+  let index = 0;
+  for (let i = 0; i < 50; i++) {
+    if (commits[i].sha.startsWith(sha)) {
+      index = i;
+      break;
+    }
+  }
+  return commits
+    .slice(0, index)
+    .map(
+      (c) =>
+        `- [${c.commit.message.split("\n\n")[0]}](${c.html_url}) by ${c.commit.author.name
+        }`
+    );
+};
+
+async function getContents(repo, path) {
+  const { data: data } = await octokit.repos.getContents({
+    owner: GH_CDS,
+    repo,
+    path,
+  });
+  return data;
+}
+
+const getHeadSha = async (repo) => {
+  const { data: repoDetails } = await octokit.repos.get({
+    owner: GH_CDS,
+    repo,
+  });
+  const { data: repoBranch } = await octokit.repos.getBranch({
+    owner: GH_CDS,
+    repo,
+    branch: repoDetails.default_branch,
+  });
+  return repoBranch.commit.sha;
+};
+
+const getLatestTag = async (repo) => {
+  const {
+    data: [latestTag],
+  } = await octokit.repos.listTags({
+    owner: GH_CDS,
+    repo,
+    per_page: 1,
+  });
+
+  return latestTag.name;
+};
+async function isNotLatestManifestsVersion() {
+  const releaseConfig = await getContents(
+    "notification-manifests",
+    "env/production/kustomization.yaml"
+  );
+
+  const releaseContent = Base64.decode(releaseConfig.content);
+  const prodVersion = releaseContent.match(
+    /notification-manifests\/\/base\?ref=(.*)/
+  )[1];
+
+  const latestVersion = await getLatestTag("notification-manifests");
+
+  return prodVersion != latestVersion;
+}
+
+async function isNotLatestTerraformVersion() {
+  const prodWorkflow = await getContents(
+    "notification-terraform",
+    ".github/workflows/infrastructure_version.txt"
+  );
+
+  const prodVersion = Base64.decode(prodWorkflow.content).trim();
+  const latestVersion = (await getLatestTag("notification-terraform")).replace(
+    "v",
+    ""
+  );
+
+  return prodVersion != latestVersion;
+}
+
+
+module.exports = { GH_CDS, AWS_ECR_URL, closePRs, createPR, getContents, getHeadSha }
```

Comment:
Is there anything that changes within these functions? or I can 🙈 on this file?

Reply:
- `createPR` has changed to loop over the files that need updating rather than just `kustomization.yaml`
- `buildLogs` has changed to uniquify the repos so that it doesn't get the same changes twice (since now we reference the api repo twice, for both k8s changes and lambda changes).